src/GraphQL.Tests/Utilities/GetGraphTypeFromTypeTests.cs

```diff
@@ -75,5 +77,127 @@ public void supports_uint_type()
+        //output mapping mode - nullable structs
+        [InlineData(typeof(int?), true, TypeMappingMode.OutputType, typeof(GraphQLClrOutputTypeReference<int>))]
+        [InlineData(typeof(decimal?), true, TypeMappingMode.OutputType, typeof(GraphQLClrOutputTypeReference<decimal>))]
+        [InlineData(typeof(Guid?), true, TypeMappingMode.OutputType, typeof(GraphQLClrOutputTypeReference<Guid>))]
+        [InlineData(typeof(MyEnum?), true, TypeMappingMode.OutputType, typeof(GraphQLClrOutputTypeReference<MyEnum>))]
+        //output mapping mode - not nullable types
+        [InlineData(typeof(int), false, TypeMappingMode.OutputType, typeof(NonNullGraphType<GraphQLClrOutputTypeReference<int>>))]
+        [InlineData(typeof(string), false, TypeMappingMode.OutputType, typeof(NonNullGraphType<GraphQLClrOutputTypeReference<string>>))]
+        [InlineData(typeof(decimal), false, TypeMappingMode.OutputType, typeof(NonNullGraphType<GraphQLClrOutputTypeReference<decimal>>))]
+        [InlineData(typeof(Guid), false, TypeMappingMode.OutputType, typeof(NonNullGraphType<GraphQLClrOutputTypeReference<Guid>>))]
+        [InlineData(typeof(Uri), false, TypeMappingMode.OutputType, typeof(NonNullGraphType<GraphQLClrOutputTypeReference<Uri>>))]
+        [InlineData(typeof(MyClass), false, TypeMappingMode.OutputType, typeof(NonNullGraphType<GraphQLClrOutputTypeReference<MyClass>>))]
+        [InlineData(typeof(MyEnum), false, TypeMappingMode.OutputType, typeof(NonNullGraphType<GraphQLClrOutputTypeReference<MyEnum>>))]
+        //output mapping mode - nullable structs
+        [InlineData(typeof(int?), false, TypeMappingMode.OutputType, null)]
+        [InlineData(typeof(decimal?), false, TypeMappingMode.OutputType, null)]
+        [InlineData(typeof(Guid?), false, TypeMappingMode.OutputType, null)]
+        [InlineData(typeof(MyEnum?), false, TypeMappingMode.OutputType, null)]
+        //output mapping mode - various types of lists
+        [InlineData(typeof(IEnumerable), true, TypeMappingMode.OutputType, typeof(ListGraphType<NonNullGraphType<GraphQLClrOutputTypeReference<object>>>))]
+        [InlineData(typeof(IEnumerable<int>), true, TypeMappingMode.OutputType, typeof(ListGraphType<NonNullGraphType<GraphQLClrOutputTypeReference<int>>>))]
+        [InlineData(typeof(IEnumerable<string>), true, TypeMappingMode.OutputType, typeof(ListGraphType<GraphQLClrOutputTypeReference<string>>))]
+        [InlineData(typeof(IEnumerable<decimal>), true, TypeMappingMode.OutputType, typeof(ListGraphType<NonNullGraphType<GraphQLClrOutputTypeReference<decimal>>>))]
+        [InlineData(typeof(IEnumerable<Guid>), true, TypeMappingMode.OutputType, typeof(ListGraphType<NonNullGraphType<GraphQLClrOutputTypeReference<Guid>>>))]
+        [InlineData(typeof(IEnumerable<Uri>), true, TypeMappingMode.OutputType, typeof(ListGraphType<NonNullGraphType<GraphQLClrOutputTypeReference<Uri>>>))]
```

Comment:
🤔 `Uri` is class so like `string` it should be nullable.

Reply:
And the case with `MyClass` too.

Reply:
>             // note: when parsing lists of class types (including lists of lists), the inner type is treated as non-nullable, except for strings


Yes, I remember...

Reply:
Ok so I'm deleting this note, and making string & MyClass so it will return a nullable list.  Correct?

Reply:
Mmmmm... it would be a "semi-breaking" change to schema: `SomeType!` -> `SomeType` for both input and output cases.
Two options:
1. Start `develop` branch and put it there.
2. Consider this as non-breaking change and keep it here.

I tend to the second option.